### PR TITLE
Remove `NativeDB::set_value` in favour of `set_values`

### PR DESCRIPTION
**Pull request by citizen-stig**
As it is only used in test.

Explicit use of `anyhow::Result` instead of importing it, I believe it improves readability.

This is ongoing effort of simplifying public interface of `sov-db` and `sov-schema-db`

### CODE REVIEW

File: full-node/db/sov-schema-db/src/lib.rs
Comment (neysofu) [inline full-node/db/sov-schema-db/src/lib.rs]: I think we should to keep this public. If private, there's no way for a downstream user to open a `DB` with custom `rocksdb::Options` and we're forcing everybody to use the options used by `DB::open`.

Comment (citizen-stig) [reply full-node/db/sov-schema-db/src/lib.rs]: public `open` also has ` db_opts: &rocksdb::Options,`, this method only provides opportunity to specify column options. 

I understand that it adds another level of configurability, but on the other hand it makes public interface more complicated. I don't have a strong evidence, that somebody might need custom options for column families. But it also does not mean, that this use case does not exist. As they say, absence of evidence is not evidence of absence.

Overall I am fine to keep it public. I will probably open follow up PR, with different approach regarding many public methods

Comment (neysofu) [reply full-node/db/sov-schema-db/src/lib.rs]: Well, apparently we do have a need for custom options (LZ4), so I think it's likely someone else will too, and it'd be nice for them to be able to use the crate. That said, I don't feel too strongly about this!

**Review by neysofu — APPROVED** [review]
LGTM! I think `open_cf` should be public but I don't want to block the PR based on that, so it's your call 😄 the rest of it seems great, especially the iterator approach to `set_values`.

File: full-node/db/sov-schema-db/src/lib.rs
Comment (citizen-stig) [reply full-node/db/sov-schema-db/src/lib.rs]: Ok, let's keep it public

**Comment by citizen-stig** [comment]
> LGTM! I think `open_cf` should be public but I don't want to block the PR based on that, so it's your call 😄 the rest of it seems great, especially the iterator approach to `set_values`.

I decided to bring it back, because I want to make this preparation work less intrusive.

**Pull request merged**
